### PR TITLE
testcase/kernel: Move user entry function for task_init to userspace

### PR DESCRIPTION
**Pull request by jeongarmy**
User entry function should be located in userspace because user can't access kernel region.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [9f48926f7c752333f654db0210785f2179de54fe] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [9f48926f7c752333f654db0210785f2179de54fe] - Code Rule Check (C++) OK.

File: apps/examples/testcase/le_tc/kernel/tc_task.c [hunk 1]
```diff
@@ -55,6 +55,7 @@ static int main_pid;
 static volatile int task_cnt;
 static volatile pid_t ppid;
 #endif
+static bool is_done;
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/kernel/tc_task.c]: Please change the name.

File: apps/examples/testcase/le_tc/kernel/tc_task.c [hunk 2]
```diff
@@ -473,9 +487,23 @@ static void tc_task_task_reparent(void)
...
+* @Scenario             :Create and Initialize a Task Control Block (TCB)
+* API's covered         :task_init
+* Preconditions         :none
+* Postconditions        :none
+* @return               :void
+*/
 static void tc_task_task_init(void)
 {
-	TC_ASSERT_EQ("task_init", ioctl(tc_get_drvfd(), TESTIOC_TASK_INIT_TEST, 0), OK);
+	TC_ASSERT_EQ("task_init", ioctl(tc_get_drvfd(), TESTIOC_TASK_INIT_TEST, test_task_entry), OK);
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/kernel/tc_task.c]: Let's initialize the is_done to false in here.

**Comment by seinfra** [comment]
Target : [c547080d125c49846f145bacf2114cb16d19ad30] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [c547080d125c49846f145bacf2114cb16d19ad30] - Code Rule Check (C++) OK.

**Pull request merged**